functions/openhab.js

```diff
@@ -141,16 +143,113 @@ function getItemsState(request,response) {
+			let currentExecutionCommand = currentCommand.execution[j];
+
+			switch (currentExecutionCommand.command) {
+				case 'action.devices.commands.OnOff':
+					turnOnOff(request, response);
+					break;
+				case 'action.devices.commands.BrightnessAbsolute':
+					adjustBrightness(request, response);
+					break;
+				case 'action.devices.commands.ChangeColor':
+				case 'action.devices.commands.ColorAbsolute':
+					adjustColor(request, response);
+					break;
+				case 'action.devices.commands.ThermostatTemperatureSetpoint':
+					adjustTemperature(request, response);
+					break;
+			}
+		}
+	}
+}
+
+/**
+ * Gets Temperature or Thermostat Data
+ */
+function getTempData(item) {
+	var thermData = {};
+	var thermItems = item.tags.toString().includes("Thermostat") ? getThermostatItems(item.members) : getThermostatItems([item]);
+
+	//Are we dealing with Fahrenheit?
+	const isF = item.tags.toString().toLowerCase().includes('fahrenheit');
+
+	//store long json variables in easier variables to work with below
+	var tstatMode = thermItems.hasOwnProperty('heatingCoolingMode') ? (thermItems.heatingCoolingMode.state.length == 1 ? utils.normalizeThermostatMode(thermItems.heatingCoolingMode.state) : thermItems.heatingCoolingMode.state) : 'heat'
+	var currTemp = thermItems.hasOwnProperty('currentTemperature') ? (isF ? utils.toC(thermItems.currentTemperature.state) : thermItems.currentTemperature.state) : '';
+	var tarTemp = thermItems.hasOwnProperty('targetTemperature') ? (isF ? utils.toC(thermItems.targetTemperature.state) : thermItems.targetTemperature.state) : '';
+	var curHum = thermItems.hasOwnProperty('currentHumidity') ? thermItems.currentHumidity.state : '';
+
+	//populate only the necessary json values, otherwise GA will get confused if keys are empty
+	if (item.tags.toString().toLowerCase().includes("thermostat")) {
+		thermData.thermostatMode = tstatMode;
+		if (thermItems.hasOwnProperty('currentTemperature')) thermData.thermostatTemperatureAmbient = Number(parseFloat(currTemp).toFixed(1));
+		if (thermItems.hasOwnProperty('targetTemperature')) thermData.thermostatTemperatureSetpoint = Number(parseFloat(tarTemp).toFixed(1));
+		if (thermItems.hasOwnProperty('currentHumidity')) thermData.thermostatHumidityAmbient = Number(parseFloat(curHum).toFixed(0));
+	}
+	else if (item.tags.toString().toLowerCase().includes("currenttemperature")) {
+		thermData.thermostatMode = "heat"; // doesn't matter, GA will only state the number (since we force Ambient and Setpoint to match)
+		if (thermItems.hasOwnProperty('currentTemperature')) thermData.thermostatTemperatureAmbient = Number(parseFloat(currTemp).toFixed(1));
+		thermData.thermostatTemperatureSetpoint = thermData.thermostatTemperatureAmbient;
+	}
+
+	return thermData;
+}
+
+/**
+ *  Retrieves Light Attributes from OpenHAB Item
+ **/
+function getLightData(item) {
+	return {
+		on: item.state === 'ON' ? true : (item.state === 0 ? false : true),
```

Comment:
This fails for me, since state could be string. See my comment: https://github.com/openhab/openhab-google-assistant/pull/24#issuecomment-368660201

Reply:
Thanks I see what you mean, if we change to Number(item.state) === 0 I have confirmed it fixes the issue.

Thanks for checking!  Do you also try the thermostat functions?  I did my best to test them manually since I only have thermostat via heat.

Reply:
No, unfortunately I don't have thermostats yet.